### PR TITLE
[RS-1991] Add enable/disable detectors doc

### PR DESCRIPTION
**Pull request by Dean-Coakley**
Product Version(s):

Calico Cloud 20.1.0 (not yet released) 

Issue:

[RS-1991](https://tigera.atlassian.net/browse/RS-1991)

Link to docs preview:

https://deploy-preview-1684--calico-docs-preview-next.netlify.app/calico-cloud/next/threat/container-threat-detection

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed 

[RS-1991]: https://tigera.atlassian.net/browse/RS-1991?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (Dean-Coakley) [inline calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx:76]: (We use `en_us` everywhere else other than this instance)

**Comment by Dean-Coakley** [comment]
Will promote to "Ready for review" after @gantony reviews content.

File: calico-cloud/threat/container-threat-detection.mdx
Comment (nelljerram) [inline calico-cloud/threat/container-threat-detection.mdx]: This is what I was talking about in stand up.  I don't think this doc makes sense, if we don't provide any guidance for how the user can infer a detector ID like "persistence-user_modification".

**Review by nelljerram — CHANGES_REQUESTED** [review]

File: calico-cloud/threat/container-threat-detection.mdx
Comment (Dean-Coakley) [reply calico-cloud/threat/container-threat-detection.mdx]: I think @gantony's suggestion of in the future generating a yaml/kubectl command should be the path forward for sure.

Are you saying the first step is not very practical/reasonable? https://github.com/tigera/docs/pull/1684/files#diff-52ad6c1409b45aadb04bb1ff6b823504bd80596585c38707317ea5d86441bae6R112-R128
If you're saying it cannot work, I'm not sure that I follow.

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (gantony) [inline calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: Would change this to `Detectors settings` to match the name of the tab, and reduce the emphasis on disabling detectors.

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (gantony) [inline calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx:29]: Not sure how to change this (and whether now is the right time), but it would be good to emphasis that these "types of suspicious activity" map to MITRE tactics https://attack.mitre.org/tactics/enterprise/

File: calico-cloud/threat/container-threat-detection.mdx
Comment (nelljerram) [reply calico-cloud/threat/container-threat-detection.mdx]: Sorry, I somehow missed the first step before.

But I still don't see how this makes sense overall.  The section is "Disable detector via RuntimeSecurity Custom Resource" and the first step is "No, disable it in the UI instead".  So what value are we actually trying to provide here?  Perhaps we should just delete this section for now?

Perhaps the point is "You've disabled some detectors in one cluster.  Now how do you use the RS resource to deploy the same to other clusters?"  That would make sense, but in that case we should update the narrative to be clearer about that being the use case.

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (gantony) [inline calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: Up to this point, a user may have no idea of what a detector is, and we want to emphasis more the "showcasing capabilities" of the feature more than disabling individual detectors.

With this in mind, I would suggest we rephrase to something like:

```
Threat detection capabilities are provided by detectors that protect against specific [MITRE techniques](https://attack.mitre.org/techniques/enterprise/). 
Detectors are grouped by [MITRE tactics](https://attack.mitre.org/tactics/enterprise/) they protect against. Additional information for each detector can be found by clicking on a detector. 
All detectors are enabled by default, with the option of disabling specific detectors if needed.

<maybe INSERT SCREENSHOT> 
```

At this stage I don't think there is much to add about `Disable detector via UI`. It's fairly self-explanatory. I've lumped it in the first section. If we feel more detail is needed, I think we should balance it with info about "view detector details". 

I would only add one sub-section for applying a configuration to multiple clusters. Maybe something along the lines of:
```
#### Configure detectors via RuntimeSecurity Custom Resource

Detector settings configured through the UI are stored in the cluster's RuntimeSecurity Custom Resource. When configuring multiple clusters, the same configuration can be applied to all clusters.

Get the reference `detectorsConfig`:

    ```
    kubectl get runtimesecurities.operator.tigera.io default -o jsonpath="{.spec.detectorsConfig}" > detectorsConfig.json
    ```

Alternatively, can enable Container Threat Detection with the required configuration in one command:
    ```
    <kubectl apply command with settings>
    ```

```

File: calico-cloud/threat/container-threat-detection.mdx
Comment (Dean-Coakley) [reply calico-cloud/threat/container-threat-detection.mdx]: My thought process was: 
The first time you do it, it's not reasonable that you'd use kubectl. So you'd disable your desired detectors via UI and then use these docs as a guide to figure out and test the commands. To make it repeatable for the rest of your clusters. - A process that we'd aim to improve in the near future.

I think it's better to leave this less-than-ideal solution documented for the time being but I'd also be ok with removing if @gantony agrees.

**Review by gantony — CHANGES_REQUESTED** [review]
I think we need to give a bit more background about what a detector is, that the main use case it to showcase them and eventually, we may disable some if needed.

Then would limit the kubectl to applying config to multiple clusters

File: calico-cloud/threat/container-threat-detection.mdx
Comment (gantony) [reply calico-cloud/threat/container-threat-detection.mdx]: Yes, I'd be in favor of providing less details about disabling detectors and only describing kubectl commands in the context of applying the same config on multiple cluster. Tried to capture this in https://github.com/tigera/docs/pull/1684#discussion_r1781042629

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (gantony) [reply calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: I initially had `#### Configure detectors via RuntimeSecurity Custom Resource` as `#### Configure detectors on multiple clusters`. Still not sure which one I prefer :sweat_smile:. The later is much more use case orientated, which might be a better fit for the docs.

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (Dean-Coakley) [reply calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx:29]: I have some unstaged changes for a following PR to update the mitre tactics at the bottom of https://docs.tigera.io/calico-cloud/threat/security-event-management#ui-help . I think it's nice that each entry references back to mitre docs on that page too.

Maybe it makes sense to reference that page from here instead as a single source of truth?

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (Dean-Coakley) [reply calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: Doesn't make much sense in the context of:
```
## How to

...
- [Detectors settings](#detectors-settings)
```

Going with `Update detectors settings`

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (gantony) [reply calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx:29]: I'm not sure. I would ignore that comment for now. I meant to revisit before pushing my review. If we mention tactics and techniques as proposed in https://github.com/tigera/docs/pull/1684#discussion_r1781042629 then I think we can leave as it for now.

I've noticed a small issue too, the tactics/categories are not order per attack phase as we do in the security events dasboard. I'll log a bug separately. I think we should rework that section of the docs only when that bug is resolved.
image

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (gantony) [reply calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: For the example, reading `detectorsConfig.json` from file could be a bit messy and distracting. We could just use and env var for simplicity and let the user figure out how to adapt it to their situation.

`detectorsConfig=$(kubectl get runtimesecurities.operator.tigera.io default -o jsonpath="{.spec.detectorsConfig}"\)`

File: calico-cloud/threat/container-threat-detection.mdx
Comment (nelljerram) [inline calico-cloud/threat/container-threat-detection.mdx]: I don't think "Alternatively" is correct here.  The next line is the second step of the process, not an alternative.

Also, don't need the `export`.  (And it looks odd to use it when not needed.)

File: calico-cloud/threat/container-threat-detection.mdx
Comment (nelljerram) [inline calico-cloud/threat/container-threat-detection.mdx]: Full stop should be a comma, i.e. these should be a single sentence.

**Review by nelljerram — APPROVED** [review]
Thanks, just some very minor tweaks from me now.

Although I've approved, I think you should wait for Antony's approval too, as he is much closer to the detail here.

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (gantony) [inline calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: I’m not convinced the 2 notes are needed now that we don’t encourage editing the CR, but just patching it.
Might be better to move them to the CR reference page added in #1685.

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (gantony) [inline calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: For the screenshot, it could be good to scroll up a tiny bit to see the 2 tabs in the page.
I like the size of the window, like we’re using a smaller screen resolution, looks good 👍

**Review by gantony — APPROVED** [review]
Looks good Dean, thanks for iterating on this.
Approving with 2 small comments, would be good to get feedback from @ctauchen too before merging.

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (Dean-Coakley) [reply calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: Good point. Updated.

File: calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx
Comment (Dean-Coakley) [reply calico-cloud_versioned_docs/version-20-1/threat/container-threat-detection.mdx]: Removed for now. Feel free to add them back in once that lands.

**Comment by ctauchen** [comment]
Closing here, commits pulled in as #1693 with edits.

**Pull request closed without merging**
